plugins/modules/mysql_db.py

```diff
@@ -386,7 +386,14 @@ def db_dump(module, host, user, password, db_name, target, all_databases, port,
             encoding=None, force=False, master_data=0, skip_lock_tables=False,
             dump_extra_args=None, unsafe_password=False, restrict_config_file=False,
             check_implicit_admin=False, pipefail=False):
-    cmd = module.get_bin_path('mysqldump', True)
+
+    if server_implementation == 'mysql':
```

Comment:
I think you'd need to adjust logic here? Ie:
```
    if server_implementation == 'mysql' or (server_implementation == 'mariadb' and LooseVersion(server_version) < LooseVersion("10.4.6")):
        cmd = module.get_bin_path('mysqldump', True)
    elif server_implementation == 'mariadb' and LooseVersion(server_version) >= LooseVersion("10.4.6"):
        cmd = module.get_bin_path('mariadb-dump', True)
```

As current will still fall under `Unknown server implementation`?

Or, detect prior version as `mysql` in here and don't bother with complex logic: https://github.com/ansible-collections/community.mysql/blob/e9845b0a1caba4344aab9e957865ac74ab17fc7f/plugins/module_utils/mysql.py#L217-L221

Reply:
I dislike the idea of setting server_implementation as "mysql" for older version of MariaDB. This could add confusion while debugging.

I pushed a version that bother about version but I think it's not that complex. What do you think @noonedeadpunk?

Reply:
oh, yes, that is much better then what I suggested :)